cura/Settings/UserChangesModel.py

```diff
@@ -0,0 +1,113 @@
+from UM.Qt.ListModel import ListModel
+
+from PyQt5.QtCore import pyqtSlot, Qt
+from UM.Application import Application
+from cura.Settings.ExtruderManager import ExtruderManager
+from UM.Settings.ContainerRegistry import ContainerRegistry
+from UM.i18n import i18nCatalog
+from UM.Settings.SettingFunction import SettingFunction
+
+import os
+
+class UserChangesModel(ListModel):
+    KeyRole = Qt.UserRole + 1
+    LabelRole = Qt.UserRole + 2
+    ExtruderRole = Qt.UserRole +3
+    OriginalValueRole = Qt.UserRole + 4
+    UserValueRole = Qt.UserRole + 6
+    CategoryRole = Qt.UserRole + 7
+
+    def __init__(self, parent = None):
+        super().__init__(parent = parent)
+        self.addRoleName(self.KeyRole, "key")
+        self.addRoleName(self.LabelRole, "label")
+        self.addRoleName(self.ExtruderRole, "extruder")
+        self.addRoleName(self.OriginalValueRole, "original_value")
+        self.addRoleName(self.UserValueRole, "user_value")
+        self.addRoleName(self.CategoryRole, "category")
+
+        Application.getInstance().globalContainerStackChanged.connect(self._update)
```

Comment:
Is this useful? 
- If the window's open, I cannot change machine
- If the window's closed, I'm not interested in the contents of the window